src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -168,99 +163,26 @@ public static boolean renameFile(Path fromFile, Path toFile, boolean replaceExis
     }
 
     /**
-     * Converts an absolute filename to a relative one, if necessary.
-     * Returns the parameter fileName itself if no shortening is possible
+     * Converts an absolute file to a relative one, if possible.
+     * Returns the parameter file itself if no shortening is possible
      * <p>
      * This method works correctly only if dirs are sorted decent in their length
      * i.e. /home/user/literature/important before /home/user/literature
      *
-     * @param fileName the filename to be shortened
-     * @param dirs     directories to check.
+     * @param file the file to be shortened
+     * @param dirs directories to check
      */
-    public static File shortenFileName(File fileName, List<String> dirs) {
-        if ((fileName == null) || !fileName.isAbsolute() || (dirs == null)) {
-            return fileName;
+    public static Path shortenFileName(Path file, List<Path> dirs) {
+        if (!file.isAbsolute()) {
+            return file;
         }
 
-        for (String dir : dirs) {
-            if (dir != null) {
-                File result = shortenFileName(fileName, dir);
-                if ((result != null) && !result.equals(fileName)) {
-                    return result;
-                }
+        for (Path dir : dirs) {
+            if (file.startsWith(dir)) {
```

Comment:
I really want to check that the `file` is a subpath of `dir`, so I think this is fine.